tools/contract-analyzer/analyzers/analyzers.go

```diff
@@ -384,6 +388,9 @@ var SupertypeInferenceAnalyzer = (func() *analysis.Analyzer {
 					case *ast.DictionaryExpression:
 						kind = "dictionaries"
 
+						_, hasExpectedType := elaboration.DictionaryExpressionHasExpectedType[element]
+						reportMissingExpectedType = !hasExpectedType
+
```

Comment:
Should a similar check be added to conditional expressions as well?

Reply:
Yes! Misread the code, it looked like expected type wasn't used there. Adding

Reply:
Just checked again, expected type is actually not used on v0.23: https://github.com/onflow/cadence/blob/v0.23/runtime/sema/check_conditional.go#L86-L87, it was only added in https://github.com/onflow/cadence/commit/8a5de647b8c69a68619c0a8880066f7a8927d0c0

I guess I can still do it though?

Reply:
Added in 967f30de1f672ff5adc5f50240eb7c99c98713c6